src/library.cpp

```diff
@@ -455,12 +455,14 @@ void Library::updateBookDB(const Book& book)
 {
   Xapian::Stem stemmer;
   Xapian::TermGenerator indexer;
-  const std::string lang = book.getLanguage();
-  try {
-    stemmer = Xapian::Stem(iso639_3ToXapian(lang));
-    indexer.set_stemmer(stemmer);
-    indexer.set_stemming_strategy(Xapian::TermGenerator::STEM_SOME);
-  } catch (...) {}
+  const auto langs = book.getLanguages();
+  if ( langs.size() == 1 ) {
+    try {
+      stemmer = Xapian::Stem(iso639_3ToXapian(langs[0]));
+      indexer.set_stemmer(stemmer);
+      indexer.set_stemming_strategy(Xapian::TermGenerator::STEM_SOME);
+    } catch (...) {}
+  }
```

Comment:
Could we try to load languages until we found one (not only checking the first one) ?

Reply:
I think you missed the `if()` surrounding the `try{}catch()`. A language (and a corresponding stemmer) is associated with the book metadata (title, description, tags, etc) only if the book is known to be in one language. Otherwise, if the book is a multilanguage one how can we figure out in which language, say, its description is written?

Reply:
I haven't missed the `if()`. But I have missed the fact it was at indexing time. I was missing with libzim feature of detect the query language based on the database language.

I was totally wrong on this side. So we are good here 🙂